### PR TITLE
Feat/xdsl emailpro

### PR DESCRIPTION
**Pull request by cbourgois**
## Add support for Email Pro in XDSL

MFRWW-38

### Related PR 

* [x] https://github.com/ovh-ux/ovh-api-services/pull/91

### CODE REVIEW

**Comment by antleblanc** [comment]
New release of [`ovh-api-services`](https://github.com/ovh-ux/ovh-api-services/releases/tag/v3.10.0) is available (`v3.10.0`).

File: client/app/telecom/pack/slots/emailPro/translations/Messages_fr_FR.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translations>
+   <translation id="email_pro_loading_error">Oups ! Nous n'avons pas pu récupérer la liste de vos XDSL E-mail.</translation>
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/translations/Messages_fr_FR.xml]: `xDSL` instead?

File: client/app/telecom/pack/slots/emailPro/pack.emailPro.constant.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+angular.module('managerApp').constant('PACK_EMAILPRO', {
+  webmailUrl: 'https://www.ovh.com/fr/mail/',
+});
+
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/pack.emailPro.constant.js]: EditorConfig rules aren't apply.

File: client/app/telecom/pack/slots/emailPro/pack-emailPro.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<div id="pack-hostedEmail"
+     data-ng-controller="PackEmailProCtrl as EmailPro">
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/pack-emailPro.html]: `as $ctrl` would be great.

File: client/app/telecom/pack/slots/emailPro/pack-emailPro.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <div class="oui-tile__body"
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/pack-emailPro.html]: Why you didn't use [`<oui-tile>`](http://master.ui-kit.ovh/#!/oui-angular/tile) here?

File: client/app/telecom/pack/slots/emailPro/detail/translations/Messages_fr_FR.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translations>
+   <translation id="email_pro_detail_loading_error" qtlid="384285">Oups ! Nous n'avons pas pu récupérer les informations relatives à votre compte email.</translation>
+   <translation id="email_pro_detail_change_password_error" qtlid="384298">Oups ! Le changement de mot de passe a échoué.</translation>
+   <translation id="email_pro_detail_change_password_success" qtlid="384311">Votre nouveau mot de passe a été modifié sur le compte {{email}}.</translation>
+   <translation id="email_pro_detail_title">XDSL E-mail {{name}}</translation>
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/detail/translations/Messages_fr_FR.xml]: `xDSL`?

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            <!-- Change password -->
+            <div class="col-md-6 col-xs-12 widget-presentation">
+                <h2 class="widget-presentation-title"
+                    data-translate="email_pro_detail_client_change_password">
+                </h2>
+                <form name=passwordChange data-ng-submit="$ctrl.changePassword()">
+                    <div
+                        data-ovh-password
+                        data-ovh-pwd="$ctrl.password"
+                        data-ovh-pwd-confirm="ctrl.account.passwordConfirmation">
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html]: `$ctrl`?

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+                    </button>
+                </form>
+            </div>
+
+            <div class="col-md-6 col-xs-12 widget-presentation">
+                <h2 class="widget-presentation-title"
+                    data-translate="email_pro_detail_client_delete_account">
+                </h2>
+
+                <div class="alert alert-warning"
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html]: Could be a [`<oui-message>`](http://master.ui-kit.ovh/#!/oui-angular/message) here I guess.

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+                        data-ovh-password
+                        data-ovh-pwd="$ctrl.password"
+                        data-ovh-pwd-confirm="ctrl.account.passwordConfirmation">
+                    </div>
+
+                    <button class="btn btn-primary"
+                            type="submit"
+                            data-ng-disabled="passwordChange.$invalid || $ctrl.changingPassword">
+                            <span data-translate="email_pro_detail_client_change_password_submit"></span>
+                            <oui-spinner data-ng-if="$ctrl.changingPassword"
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html]: See https://github.com/ovh-ux/ovh-manager-telecom/pull/756/files#diff-2fbd95b44257cc5fabd8ec6c43c704beR129.

You have a spinner and text inside your `<button>` element. Here is the opposite.

File: client/app/telecom/pack/slots/emailPro/add/translations/Messages_fr_FR.xml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translations>
+   <translation id="add_an_account">Ajouter un compte XDSL E-mail</translation>
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/translations/Messages_fr_FR.xml]: `xDSL`?

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+angular.module('managerApp').config(($stateProvider) => {
+  $stateProvider.state('telecom.pack.emailPro-add', {
+    url: '/xdsl-email/add',
+    views: {
+      'packView@telecom.pack': {
+        templateUrl: 'app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html',
+        controller: 'PackEmailProAddCtrl',
+        controllerAs: 'ctrl',
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.js]: `$ctrl` would be great.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 11]
```diff
@@ -0,0 +1,88 @@
+<div class="telecom-pack-hostedEmail-add">
+
+    <header>
+        <section-back-link section-back-link-to-state="telecom.pack"></section-back-link>
+        <h1 data-translate="add_an_account"></h1>
+    </header>
+
+
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: Useless empty line.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+                <h2 class="widget-presentation-title"
+                    data-translate="account_informations">
+                </h2>
+
+                <span data-translate="all_fields_are_required"></span>
+
+                <form name="accountForm"
+                      data-ng-submit="ctrl.add()">
+                    <div class="form-group">
+                        <div class="row">
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: `.row > .col-xs-12` can be a used as a single `<div>`.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+                            <div class="col-xs-12 input-select">
+                                <div>
+                                    <label class="control-label" data-translate="account_name"></label>
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: Link between `for=""` and `id=""`on your `<input>` element.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+                                    <div class="row">
+                                        <div class="col-xs-10">
+                                            <div class="input-select-container">
+                                                <div class="input-group">
+                                                    <input name="accountName"
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: Missing `type="??"`.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+                                                            data-ng-model="ctrl.account.domain"
+                                                            data-ng-options="item.value as item.label for item in ctrl.availablesDomains"
+                                                            required>
+                                                    </select>
+                                                </div>
+                                            </div>
+                                        </div>
+                                        <div class="col-xs-2">
+                                            <h5 data-ng-show="accountForm.accountName.$dirty
+                                                        && accountForm.accountName.$valid
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: I didn't reach any `<h3>` and `<h4>` tags. What's the reason of using an `<h5>` tag?

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+                                                        && accountForm.accountDomain.$dirty
+                                                        && accountForm.accountDomain.$valid"
+                                                  class="ovh-font ovh-font-success text-success"
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: Can we avoid `ovh-font-*`? They are deprecated.

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+                                                            class="form-control"
+                                                            data-ng-model="ctrl.account.domain"
+                                                            data-ng-options="item.value as item.label for item in ctrl.availablesDomains"
+                                                            required>
+                                                    </select>
+                                                </div>
+                                            </div>
+                                        </div>
+                                        <div class="col-xs-2">
+                                            <h5 data-ng-show="accountForm.accountName.$dirty
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.html]: Prefer `ngIf`?

File: client/app/telecom/pack/pack.js [hunk 18]
```diff
@@ -18,9 +18,11 @@ angular.module('managerApp').config(($stateProvider) => {
       $title(translations, $translate, OvhApiPackXdsl, $stateParams) {
         return OvhApiPackXdsl.v6().get({
           packId: $stateParams.packName,
-        }).$promise.then(data => $translate.instant('pack_xdsl_page_title', { name: data.description || $stateParams.packName }, null, null, 'escape')).catch(() => $translate('pack_xdsl_page_title', { name: $stateParams.packName }));
+        }).$promise
+          .then(data => $translate.instant('pack_xdsl_page_title', { name: data.description || $stateParams.packName }, null, null, 'escape'))
+          .catch(() => $translate('pack_xdsl_page_title', { name: $stateParams.packName }));
```
Comment (antleblanc) [inline client/app/telecom/pack/pack.js]: `.instant(` is missing?

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.controller.js [hunk 19]
```diff
@@ -0,0 +1,75 @@
+angular.module('managerApp').controller('PackEmailProAddCtrl', class PackEmailProAddCtrl {
+  constructor($q, $location, $scope, $state, $stateParams, $translate, OvhApiPackXdsl, Toast) {
+    this.$q = $q;
+    this.$location = $location;
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.controller.js]: -`this.$location`

File: client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.controller.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+  add() {
+    this.pendindOrder = true;
+
+    return this.OvhApiPackXdsl.EmailPro().v6()
+      .save({ packName: this.$stateParams.packName }, {
+        email: this.account.name + this.account.domain,
+        password: this.account.password,
+      }).$promise
+      .then(() => this.$state.go('telecom.pack', { packName: this.$stateParams.packName }))
+      .then(() => {
```
Comment (antleblanc) [inline client/app/telecom/pack/slots/emailPro/add/pack-emailPro-add.controller.js]: You can remove `{`.

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.controller.js [hunk 21]
```diff
@@ -0,0 +1,98 @@
+angular.module('managerApp').controller('PackEmailProDetailCtrl', class PackEmailProDetailCtrl {
+  constructor(
+    $http, $q, $stateParams, $state, $translate,
+    OvhApiPackXdsl, OvhApiXdsl, PACK_EMAILPRO, Toast, URLS,
```
Comment (marie-j) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.controller.js]: The params are passed in an order different than it is done in `pack.controller.js`. We should probably keep the same order everywhere

**Comment by ovh-cds** [comment]
CDS Report build#6105.0 ✘  
* init
  * deps ✘

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.controller.js (hunk 21 above)
Comment (cbourgois) [reply client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.controller.js]: Alphabetical Order ($ first)

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html [hunk 22]
```diff
@@ -0,0 +1,139 @@
+<section class="telecom-section">
+    <div class="telecom-section-header">
+        <header role="banner">
+            <section-back-link section-back-link-to-state="telecom.pack"
+                               section-back-link-title="{{ $ctrl.pack.description || $ctrl.pack.packName }}">
+            </section-back-link>
+            <h1 data-translate="email_pro_detail_title"
+                data-translate-values="{name: $ctrl.serviceName}"></h1>
```
Comment (JeremyDec) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html]: `{ name: $ctrl.serviceName }` (missing spaces after opening bracket and closing one)

File: client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html [hunk 23]
```diff
@@ -0,0 +1,139 @@
+<section class="telecom-section">
+    <div class="telecom-section-header">
+        <header role="banner">
+            <section-back-link section-back-link-to-state="telecom.pack"
+                               section-back-link-title="{{ $ctrl.pack.description || $ctrl.pack.packName }}">
```
Comment (JeremyDec) [inline client/app/telecom/pack/slots/emailPro/detail/pack-emailPro-detail.html]: `{{:: $ctrl.pack.description || $ctrl.pack.packName }}` is possible no ?

**Pull request merged**